inventory/inv_scripts/inventory_ui.gd

```diff
@@ -0,0 +1,31 @@
+extends Control
+
+@onready var inv: Inv = preload("res://inventory/player_inventory.tres")
+@onready var slots: Array = $NinePatchRect/GridContainer.get_children()
+
+
+var is_open = false
+
+func _ready():
+	inv.update.connect(update_slots)
+	update_slots()
+	close()
+	
+func update_slots():
+	for i in range(min(inv.slots.size(), slots.size())):
+		slots[i].update(inv.slots[i])
+
+func _process(delta):
+	if Input.is_action_just_pressed("i"):
```

Comment:
Check `player.gd` for how it handles input. You can define custom actions in Godot -> Project -> Inputs (or something like that). Add an input `inventory` and bind it to "i". That way, we can very easily implement key remapping in the future instead of hardcoding keys like this.